scripts/typesFiles.js

```diff
@@ -0,0 +1,153 @@
+#!/usr/bin/env node
+const fs = require("fs");
+// eslint-disable-next-line import/no-extraneous-dependencies
+const shell = require("shelljs");
+// eslint-disable-next-line import/no-extraneous-dependencies
+const reactDocs = require("react-docgen");
+
+const skipPackages = [
+  "Guard",
+  "Icon",
+  "Stylers",
+  "Tokens",
+  "helpers",
+  "Button",
+  "ButtonGroup",
+  "Calendar",
+  "DataGrid",
+  "DateInput",
+  "Heading",
+  "Input",
+  "ListBox",
+  "Modal",
+  "Popover",
+  "RawButton",
+  "Select",
+  "Textarea",
+  "Toast",
+  "Uploader",
+];
+
+const fileName = "index.d.ts";
+/* prettier-ignore */
+const renderDeclarationTemplate = ({ displayName="", props = "" }) => { return `export default ${displayName};
+${props}`;};
+
+const createPropsList = ({ info }) => {
+  if (!info || !info.props) return "";
+  const subComponent = info.displayName.substring(info.displayName.indexOf(".") + 1, info.displayName.length);
+  const displayName = info.displayName.includes(".") ? subComponent : info.displayName;
+  /* prettier-ignore */
+  const list = [`
+declare function ${displayName}(props: any): JSX.Element;
```

Comment:
if we already know which are the props we can declare the props instead of passing any:
using this:
```js
  Object.keys(info.props).map(key => {
```

the template would look like:
```js
interface IPRops {
    kind: "primary" | "secondary",
    size?: "medium" | "large",
}
```

we can even add the `?` if the prop is optional or without it if it's required. 

Maybe this is a stretch and the next step, but this could improve the developer experience.

Reply:
`?` for optional props has been added, but still need to check some of the  generated props value in the `d.t.s` file